examples/AddressBook/CMakeLists.txt

```diff
@@ -1,6 +1,11 @@
 
 facelift_add_interface(AddressBookInterface
     INTERFACE_DEFINITION_FOLDER ${CMAKE_CURRENT_SOURCE_DIR}/interface
+    HEADERS_GLOB_RECURSE_NO_MOC 
```

Comment:
users of facelift_add_interface do not know which files are generated. They know even less about which files need to be MOCed.

Reply:
It seems that the same files are generating the warnings
* Module.h
* ModuleIPC.h
* ModulePrivate.h
* ObjectRegistryIPCCommon.h
* {interface_name}IPCCommon.h
     e.g. AddressBookIPCCommon.h and SubInterfaceIPCCommon.h

How are you see if we'll create some moc blacklist for such files 
instead of adding them to each CMakeLists.txt ?

Reply:
you can let the code generator generate a new file containing the following:
_set(GENERATED_HEADERS_MOC
    File1.h
    File2.h
    ....
)
set(GENERATED_HEADERS_NO_MOC
    File3.h
    File4.h
    ....
)
set(GENERATED_SOURCES
    File1.cpp
    File2.cpp
    ....
)_

In order to identify which files should be MOCed, the code generator can search for "Q_OBJECT" in the files.
Then, the implementation of the "facelift_add_interface" function can use the "include" command (https://cmake.org/cmake/help/v3.0/command/include.html) to get the definitions of those variables and call the facelift_add_library with those file lists.

Reply:
I 100% agree with the first part, about generating config files. We know, during the code generation, which files require running moc. 

However searching for Q_OBJECT is most probably wrong. We would have to look also for Q_GADGET and probably a few other macros. Someone could also define Q_OBJECT inside other macro (not applicable to generated code) and we may miss it. Another point: this is slower for generated code, for which we already know if we have to run moc. I would suggest to keep it to developer decision. 

This way or another, user should not add generated headers to HEADERS_NO_MOC. The user should add only his headers. This applies to all macros.

Reply:
one problem that our templates can generate or not Q_OBJECT macro.
e.g. build/facelift_generated/MyPackageInterface/module/facelift/example/mypackage/ModulePrivate.h doesn't have Q_OBJECT
but 
build/facelift_generated/MyPackageInterface/module/facelift/example/mypackage/ModulePrivate.h
does not
because codegen/facelift/templates/ModulePrivate.template.h 
has for-loop in which we can omit Q_OBJECT generating.

so probably Generator must search for some of macros, 
but those macros could be defined in CMAKE_AUTOMOC_MACRO_NAMES (https://cmake.org/cmake/help/v3.12/variable/CMAKE_AUTOMOC_MACRO_NAMES.html#variable:CMAKE_AUTOMOC_MACRO_NAMES)

that's why we should search for Q_OBJECT etc. via cmake not in Generator

Reply:
I've just forgot that we do not use AUTOMOC...

so we can define CMAKE_MOC_MACRO_NAMES as follows
   
set(CMAKE_MOC_MACRO_NAMES "Q_OBJECT" "Q_GADGET" "Q_NAMESPACE")

btw. why do not we use AUTOMOC?